### PR TITLE
Dataset Evaluation Steps

### PR DESCRIPTION
**Pull request by adason**
I have created this draft review as a starting point for the step-by-step guide using dataset insights/kubeflow pipelines for dataset evaluation. Most of the section are left blink and I will have to gradually fill them in.

### CODE REVIEW

File: docs/DatasetEvaluation.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+TODO: check if the notebook contain performance metrics
+
+### Using our pre-trained models
+We trained a model using ~400k synthetic examples and then fine tuned it using ~700 real images. You can use the same [visual inspection notebook] mentioned above, but use one of our models from the list below:
+
+- [Real World (760)](http://url)
+- [Synthetic (400,000)](http://url)
+- [Synthetic (400,000) + Real World (76)](http://url)
+- [Synthetic (400,000) + Real World (380)](http://url)
+- [Synthetic (400,000) + Real World (760)](http://url)
```
Comment (BlairLee) [inline docs/DatasetEvaluation.md:146]: How could I add the public model URL in here?

File: docs/DatasetEvaluation.md [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+
+Simply import the [pre-compiled pipeline](https://raw.githubusercontent.com/Unity-Technologies/datasetinsights/master/kubeflow/compiled/evaluate_the_model.yaml) into your kubeflow cluster. The figure below shows how to do this using the web UI. You can optionally use the [KFP CLI Tool](https://www.kubeflow.org/docs/pipelines/sdk/sdk-overview/#kfp-cli-tool)
+
+![upload pipeline](images/kubeflow/upload_pipeline.png)
+
+Once your pipeline has been imported, you can run it vis the web UI as shown below. Alternatively, you can use the [KFP CLI Tool](https://www.kubeflow.org/docs/pipelines/sdk/sdk-overview/#kfp-cli-tool)
+
+![evaluate the model](images/kubeflow/evaluate_the_model.png)
+
+Whether you trained a model on synthetic, real or multiple datasets, you'll need to specify a model stored in the `checkpoint_dir` from previous pipelines.
```
Comment (BlairLee) [inline docs/DatasetEvaluation.md:184]: Can you tell users how to select a model (e.g. select the best model based on validation loss)?

File: docs/DatasetEvaluation.md [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+You have to specify run parameters required by this pipeline:
+
+- `docker`: Path to a Docker Registry. We suggest changing this parameter to pull our images on Docker Hub with a specific tag, such as `unitytechnologies/datasetinsights:0.2.0`
+- `source_uri`: The dataset source uri. You can use the default value which points to the required dataset for this pipeline.
+- `config`: Estimator config YAML file. You can use the default value which points to a YAML file packaged with our docker images.
+- `checkpoint_file`: Path to the Estimator checkpoint file from previous training runs that you want to load for evaluation.
+- `tb_log_dir`: Path to store tensorboard logs used to visualize the training progress.
+- `volume_size`: Size of the Kubernetes Persistent Volume Claims (PVC) that will be used to store the dataset. You can use the default value.
+
+Just like for the training pipeline, you'll want to change `tb_log_dir` to point to a location that is convenient for you and your have permission to write to. This is where you'll read the logs and see the **performance metrics** once the pipeline completes.
```
Comment (BlairLee) [inline docs/DatasetEvaluation.md:195]: Typo. "your have permission" should be "you have permission"

File: docs/DatasetEvaluation.md [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+
+In addition to the logs, the performance metrics are also available in a Jupyter Notebook we have prepared that includes code to visualize the predictions.
+
+### visualizing predictions and performance
+
+We recommend running our [docker image](https://hub.docker.com/r/unitytechnologies/datasetinsights) which includes Jupyter as well as our notebooks if you don't want to setup the environment on your own. You can follow the same [instructions](https://github.com/Unity-Technologies/SynthDet/blob/master/docs/RunningSynthDetCloud.md#step-6-run-dataset-statistics-using-the-datasetinsights-jupyter-notebook) you may have used to run our [statistics notebook](https://github.com/Unity-Technologies/datasetinsights/blob/master/notebooks/SynthDet_Statistics.ipynb), but select the newer [visual inspection notebook](https://github.com/Unity-Technologies/datasetinsights/blob/master/notebooks/SynthDet_Evaluation.ipynb) called `SynthDet_Evaluation.ipynb` instead.
```
Comment (BlairLee) [inline docs/DatasetEvaluation.md:201]: Why do we need to enhance the "newer" notebook?

File: docs/DatasetEvaluation.md [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+
+### Using our pre-trained models
+
+We trained a model using `~400k` synthetic examples and then fine tuned it using `~700` real images. You can use the same [visual inspection notebook](https://github.com/Unity-Technologies/datasetinsights/blob/master/notebooks/SynthDet_Evaluation.ipynb) mentioned above, but use one of our models from the list below:
```
Comment (BlairLee) [inline docs/DatasetEvaluation.md:205]: Typo. "fine-tuned"

File: docs/DatasetEvaluation.md (hunk 4 above)
Comment (adason) [reply docs/DatasetEvaluation.md:201]: removed the word `newer` as it's redundant

File: docs/DatasetEvaluation.md (hunk 2 above)
Comment (adason) [reply docs/DatasetEvaluation.md:184]: The selection of "select the best model based on validation loss" is already covered in various training sections. Here we simply want to focus on "evaluate a model" which is independent of how they pick a model.

**Comment by BlairLee** [comment]
Please make sure that all the typos are solved. E.g. `evaluate evaluate` should be `evaluate`.

File: docs/DatasetEvaluation.md [hunk 6]
```diff
@@ -0,0 +1,263 @@
...
+
+To train the model, simply [import](https://www.kubeflow.org/docs/pipelines/pipelines-quickstart/#deploy-kubeflow-and-open-the-pipelines-ui) the pre-compiled [pipeline](https://raw.githubusercontent.com/Unity-Technologies/datasetinsights/master/kubeflow/compiled/train_on_synthdet_sample.yaml) into your kubeflow cluster. The figure below shows how to do this using the web UI. You can optionally use the [KFP CLI Tool](https://www.kubeflow.org/docs/pipelines/sdk/sdk-overview/#kfp-cli-tool)
+
+![upload pipeline](images/kubeflow/upload_pipeline.png)
+
+Once your pipeline has been imported, you can run it via the web UI as shown below. Alternatively, you can use the [KFP CLI Tool](https://www.kubeflow.org/docs/pipelines/sdk/sdk-overview/#kfp-cli-tool)a
+![train on SynthDet sample](images/kubeflow/train_on_synthdet_sample.png)
```
Comment (sanjayuconn) [inline docs/DatasetEvaluation.md:38]: typo 'a': shows "KFP CLI Toola"

File: docs/DatasetEvaluation.md [hunk 7]
```diff
@@ -0,0 +1,263 @@
...
+
+Next, follow the [instructions](#part-3-evaluate-a-model) to evaluate evaluate the performance of this model by running one more pipeline we have prepared. You'll need the location of your model in the next step.
+
+## Part 3: Evaluate a model
+
+In this section, we'll a trained model to generate predictions on a held out test set of real images and measure its performance using well-known performance metrics like [mAP](https://datasetinsights.readthedocs.io/en/latest/datasetinsights.evaluation_metrics.html#datasetinsights.evaluation_metrics.average_precision_2d.MeanAveragePrecisionAverageOverIOU) and [mAR](https://datasetinsights.readthedocs.io/en/latest/datasetinsights.evaluation_metrics.html#datasetinsights.evaluation_metrics.average_recall_2d.MeanAverageRecallAverageOverIOU). We have prepared another kubeflow [pipeline](https://raw.githubusercontent.com/Unity-Technologies/datasetinsights/master/kubeflow/compiled/evaluate_the_model.yaml) for this.
```
Comment (sanjayuconn) [inline docs/DatasetEvaluation.md:201]: Opening sentence could be better: specifically this part: "we'll a trained model"

File: docs/DatasetEvaluation.md [hunk 8]
```diff
@@ -0,0 +1,263 @@
+# Dataset Evaluation
+
+This guide shows you how to evaluate the value/quality of a synthetic dataset by training a [Faster-RCNN](https://arxiv.org/abs/1506.01497) object detection model on it and testing the performance of that model on a well-known held out dataset of real images
+
```
Comment (sanjayuconn) [inline docs/DatasetEvaluation.md:4]: - We can have list of topics in the beginning of this page.
- We may want to take common steps out of every pipeline testing process and refer there as needed.

File: docs/DatasetEvaluation.md (hunk 6 above)
Comment (adason) [reply docs/DatasetEvaluation.md:38]: fixed

File: docs/DatasetEvaluation.md (hunk 7 above)
Comment (adason) [reply docs/DatasetEvaluation.md:201]: Changed to:
>In this section, we'll use a trained model to generate predictions on the test split of UnityGroceries-RealWorld dataset and measure its performance using well-known performance metrics like

File: docs/DatasetEvaluation.md (hunk 8 above)
Comment (adason) [reply docs/DatasetEvaluation.md:4]: The common steps are repeated on-purpose to make sure readers does not have to jump between sections which breaks the flow of following this instructions. Users are not expected to go through every sections of this document. Instead, they are encouraged to skip "(optional)" sections.

What do you mean by list of topics in the beginning of this page? Do you mean something like table of content that links to each section?

**Comment by adason** [comment]
This documentation will be moved to https://github.com/Unity-Technologies/datasetinsights/pull/84. Closing this review.
